server/src/main/java/org/apache/kafka/security/authorizer/AclEntry.java

```diff
@@ -55,96 +44,17 @@
-    private static final String PERMISSION_TYPE_KEY = "permissionType";
-    private static final String OPERATION_KEY = "operation";
-    private static final String HOSTS_KEY = "host";
-    public static final String VERSION_KEY = "version";
-    public static final int CURRENT_VERSION = 1;
-    private static final String ACLS_KEY = "acls";
-
-    public final AccessControlEntry ace;
-    public final KafkaPrincipal kafkaPrincipal;
-
-    public AclEntry(AccessControlEntry ace) {
-        super(ace.principal(), ace.host(), ace.operation(), ace.permissionType());
-        this.ace = ace;
-
-        kafkaPrincipal = ace.principal() == null
-            ? null
-            : SecurityUtils.parseKafkaPrincipal(ace.principal());
-    }
-
-    /**
-     * Parse JSON representation of ACLs
-     * @param bytes of acls json string
-     *
-     * <p>
-        {
-            "version": 1,
-            "acls": [
-                {
-                    "host":"host1",
-                    "permissionType": "Deny",
-                    "operation": "Read",
-                    "principal": "User:alice"
-                }
-            ]
-        }
-     * </p>
-     *
-     * @return set of AclEntry objects from the JSON string
-     */
-    public static Set<AclEntry> fromBytes(byte[] bytes) throws IOException {
-        if (bytes == null || bytes.length == 0)
-            return Collections.emptySet();
-
-        Optional<JsonValue> jsonValue = Json.parseBytes(bytes);
-        if (jsonValue.isEmpty())
-            return Collections.emptySet();
-
-        JsonObject js = jsonValue.get().asJsonObject();
-
-        //the acl json version.
-        Utils.require(js.apply(VERSION_KEY).to(INT) == CURRENT_VERSION);
-
-        Set<AclEntry> res = new HashSet<>();
-
-        Iterator<JsonValue> aclsIter = js.apply(ACLS_KEY).asJsonArray().iterator();
-        while (aclsIter.hasNext()) {
-            JsonObject itemJs = aclsIter.next().asJsonObject();
-            KafkaPrincipal principal = SecurityUtils.parseKafkaPrincipal(itemJs.apply(PRINCIPAL_KEY).to(STRING));
-            AclPermissionType permissionType = SecurityUtils.permissionType(itemJs.apply(PERMISSION_TYPE_KEY).to(STRING));
-            String host = itemJs.apply(HOSTS_KEY).to(STRING);
-            AclOperation operation = SecurityUtils.operation(itemJs.apply(OPERATION_KEY).to(STRING));
-
-            res.add(new AclEntry(new AccessControlEntry(principal.toString(),
-                host, operation, permissionType)));
-        }
-
-        return res;
-    }
-
-    public static Map<String, Object> toJsonCompatibleMap(Set<AclEntry> acls) {
-        Map<String, Object> res = new HashMap<>();
-        res.put(AclEntry.VERSION_KEY, AclEntry.CURRENT_VERSION);
-        res.put(AclEntry.ACLS_KEY, acls.stream().map(AclEntry::toMap).collect(Collectors.toList()));
-        return res;
+    
+    public AclEntry(String principal, String host, AclOperation operation, AclPermissionType permissionType) {
```

Comment:
Do we need this constructor? It seems it's only used by `AuthorizerBenchmark` which in turn uses it to call the `AclBinding` constructor, but that constructor takes a `AccessControlEntry` instance. So we should be able to delete it.

Reply:
agree to @mimaison 

@m1a2st could you please refactor `AuthorizerBenchmark` to use `AccessControlEntry` instead of `AclEntry`? With that refactor, in codebase we don't have any usage of constructor of `AclEntry`, and hence we can do more cleanup for `AclEntry`